LanguageFeatures/Extension-types/exhaustiveness_map_A01_t01.dart

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Switch statements and expressions with map as a matched type can
+/// never be exhaustive
```

Comment:
I couldn't find this claim in the feature specification, and it wouldn't be true, anyway: The matched value type can be of the form `Map<K, V>` for some types `K` and `V`, and we can easily have this and be exhaustive:

```dart
void main() {
  var x = switch (<int, String>{...}) {
    _ => 1,
  };
}
```

If there is a non-extension-type version of this test then a similar assertion/description would presumably need an extra look.

The `@description` is quite different, it talks about a map pattern (the matched value type is on one side of the fence, each pattern is on the other side). So the question is: Can we write a map pattern that successfully matches an arbitrary matched object?

I think you're right that there is no single pattern which will match all maps of any given type: We can't use an ellipsis as a "rest pattern" or anything like that, and we cannot match an empty map using a map pattern.

So what's the conclusion? Probably something like "exhaustiveness with map patterns can only be achieved when there is an exhaustive pattern in addition to any map patterns", like:

```dart
void main() {
  var x = switch (<int, String>{...}) {
    {1: ''} => 1,
    Map() => 2,
  };
}
```

Alternatively, we could try some set of switches whose cases are all map patterns, and see that the result isn't exhaustive. However, that's a very long shot, because we'd need to argue that "we've tried all possibilities!" Otherwise there isn't much that we can conclude based on the result "this one isn't exhaustive; that one neither; or that one; ...".